utilities/include/vec4.hpp

```diff
@@ -0,0 +1,171 @@
+#ifndef VEC4_HPP
+#define VEC4_HPP
+
+#include <cstddef>
+#include <ostream>
+#include <cmath>
+#include <xsimd/xsimd.hpp>
+#include <type_traits>
+
+#include "config.hpp"
+
+/*  *Macros provided by config.hpp
```

Comment:
It would make more sense to put this comment on top of the `#include "config.hpp` line.